charts/kafka-topics/templates/topics.yaml

```diff
@@ -0,0 +1,25 @@
+{{- $root := . -}}
+{{- range .Values.topics }}
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaTopic
+metadata:
+  name: {{ .name }}
+  {{- if $root.Values.namespace }}
+  namespace: {{ $root.Values.namespace }}
+  {{- end }}
+  labels:
+    strimzi.io/cluster: {{ $root.Values.strimziCluster }}
+    {{- range $key, $value := $root.Values.labels }}
+    {{ $key }}: {{ $value }}
+    {{- end }}
+    {{ include "kafka-topics.labels" $ | nindent 4 }}
+spec:
+  {{- toYaml .spec | nindent 2 }}
+  {{- if and (not .spec.partitions) ($root.Values.partitions) }}
```

Comment:
For every topic in spec you can configure everything Strimzi Topic Operator supports, e.g.:
```
topics:
- name: example-topic
  spec:
    partitions: 1
    replicas: 1
    config:
      retention.ms: 7200000
      segment.bytes: 1073741824
```

You can set the number of partitions and/or replicas in the root of Values or per topic in the spec. If you define it in the spec for every topic this will always overwrite the number set in the root

Reply:
Good point, I added it. But it will not merge the config in root and the one defined per topic. So if `spec.config` is defined for a topic the `.Values.config` configuration will be ignored for this topic.